src/sentry/static/sentry/app/views/ruleEditor/index.jsx

```diff
@@ -158,9 +159,7 @@ const RuleEditor = createReactClass({
     // If 'All Environments' is selected remove the environment property from the data
     if (val === 'all') {
       this.setState(state => {
-        const rule = {...state.rule};
-        delete rule.environment;
-        return {rule};
+        return {rule: state.rule};
```

Comment:
Do we need a re-render here? new state looks to be the same as prev state. I would add a comment to clarify the intention of this `setState` since it's not clear

Reply:
Thanks, yeah that's unnecessary